### ISSUE TITLE
RFE: Macro feature for saving and reusing input to send to terminal session

### ISSUE THREAD

**Issue by Conan-Kudo** [gnunn1/terminix#88]
One particularly interesting feature that I've seen in Xshell for Windows is the ability to record scripts, commands, etc. as a key combination or a button that can clicked to insert and run in a given terminal session (or all of them).

The feature is called Quick Command in Xshell, and it's briefly described on their product's [key features page](https://www.netsarang.com/products/xsh_key_features.html).

It'd be awesome if Terminix supported something similar.

**Comment by Asher256** [comment]
+1

**Comment by gnunn1** [comment]
If someone could point me at some screenshots of this feature implemented elsewhere or do a mockup it would be quite helpful.

**Comment by Asher256** [comment]
Another nice feature: the possibility to script terminix.

For example a script that will do this automatically: create a 2x2 Terminix splits and run "ssh host1" in the first split, "ssh host2" in the second split, vim file.cpp in the third split, etc. 

The scripting language could be Lua or Python for example.

The scripting feature will be appreciated by programmers and Linux administrators because they could have 1 script per project (and each project will spawn automatically and create a Terminix view for each project / group of servers).

**Comment by gnunn1** [comment]
@Asher256 You can do that already with a session file and for each terminal overriding the command to execute in the terminal.

**Comment by Asher256** [comment]
Does the "session" feature manage the splits and each command I can run on splits? (like 2 horizontal splits + 3 vertical splits on each horizontal split)

Can you tell me more about this session feature? Where can I go modify it?

**Comment by gnunn1** [comment]
Yes, when you save the session file it saves the placement of all of the individual terminals in the session. You can then load the session from the GUI, via the Load menu item, or the command line using the --session switch.

To override the command that is used for each terminal, instead of just loading a shell, go into each terminal menu and select 'Layout Options..'. Enter the command you want that terminal to load in the Command field. Repeat this process for each terminal in turn and then save the session file. Next time you load the session file it should run these commands.

**Comment by gnunn1** [comment]
This would be pretty feasible using LUA as there is an existing D integration with it. After that, it's just a matter of designing some interfaces to expose to LUA.

https://github.com/JakobOvrum/LuaD

**Comment by yodatak** [comment]
It could be simple now to use something like the password prompt and asign they to a shortcut , like doing a sudo or something
